tool/tsh/kubectl.go

```diff
@@ -0,0 +1,272 @@
+func runKubectlCode(args []string) {
+	// These values are the defaults used by kubectl and can be found here:
+	// https://github.com/kubernetes/kubectl/blob/3612c18ed86fc0a2f4467ca355b3e21569fabe0a/pkg/cmd/cmd.go#L94
+	defaultConfigFlags := genericclioptions.NewConfigFlags(true).
+		WithDeprecatedPasswordFlag().
+		WithDiscoveryBurst(300).
+		WithDiscoveryQPS(50.0)
+
+	command := cmd.NewDefaultKubectlCommandWithArgs(
+		cmd.KubectlOptions{
+			// init the default plugin handler.
+			PluginHandler: cmd.NewDefaultPluginHandler(plugin.ValidPluginFilenamePrefixes),
+			Arguments:     args,
+			ConfigFlags:   defaultConfigFlags,
+			// init the IOSStreams.
+			IOStreams: genericclioptions.IOStreams{In: os.Stdin, Out: os.Stdout, ErrOut: os.Stderr},
+		},
+	)
+	// override args without kubectl to avoid errors.
+	command.SetArgs(args[1:])
+	// run command until it finishes.
+	if err := cli.RunNoErrOutput(command); err != nil {
+		// Pretty-print the error and exit with an error.
+		cmdutil.CheckErr(err)
+	}
+	os.Exit(0)
+}
+
+func runKubectlAndCollectRun(cf *CLIConf, args []string) error {
+	var (
+		alreadyRequestedAccess bool
+		err                    error
+		exitErr                *exec.ExitError
+	)
+	for {
+		// missingKubeResources will include the Kubernetes Resources whose access
+		// was rejected in this kubectl call.
+		missingKubeResources := make([]resourceKind, 0, 50)
+		reader, writer := io.Pipe()
+		wg := sync.WaitGroup{}
+		wg.Add(1)
+		go func() {
+			// This goroutine scans each line of output emitted to stderr by kubectl
+			// and parses it in order to check if the returned error was a problem with
+			// missing access level. If it's the case, tsh kubectl will create automatically
+			// the access request for the user to access the resource.
+			// Current supported resources:
+			// - pod
+			// - kube_cluster
+			defer wg.Done()
+
+			scanner := bufio.NewScanner(reader)
+			scanner.Split(bufio.ScanLines)
+			for scanner.Scan() {
+				line := scanner.Text()
+
+				// Check if the request targeting a pod endpoint was denied due to
+				// Teleport Pod RBAC or if the operation was denied by Kubernetes RBAC.
+				// In the second case, we should create a Resource Access Request to allow
+				// the user to exec/read logs using different Kubernetes RBAC principals.
+				// using different Kubernetes RBAC principals.
+				if podForbiddenRe.MatchString(line) {
+					results := podForbiddenRe.FindStringSubmatch(line)
+					missingKubeResources = append(missingKubeResources, resourceKind{kind: types.KindKubePod, subResourceName: filepath.Join(results[2], results[1])})
+					// Check if cluster access was denied. If denied we should create
+					// a Resource Access Request for the cluster and not a pod.
+				} else if strings.Contains(line, clusterForbidden) || clusterObjectDiscoveryFailed.MatchString(line) {
+					missingKubeResources = append(missingKubeResources, resourceKind{kind: types.KindKubernetesCluster})
+				}
+			}
```

Comment:
I think you also need to check for `scanner.Err()`.

Reply:
Added check for `scanner.Err`